File: 06-customCustoms/python/partOne.py
from common import *


class Group:
    questions: List[str]
    num_pax: int

    def __init__(self, instr: str) -> None:
        individual_pax = instr.split("\n")
        self.num_pax = len(individual_pax)

        self.questions = []

        for pax in individual_pax:
            for char in pax:
                if char not in self.questions:
                    self.questions.append(char)


def partOne(instr: str) -> int:
    groups = [Group(x) for x in parse(instr)]

    question_total = 0

    for group in groups:
        question_total += len(group.questions)

    return question_total

File: challenges/2020/06-customCustoms/go/challenge/partOne.go
package challenge

import "strings"

func PartOne(instr string) int {
	newGroup := func(instr string) Group {
		g := Group{}

		individualPax := strings.Split(instr, "\n")
		g.NumPax = len(individualPax)

		for _, pax := range individualPax {
			for _, char := range pax {
				if !IsRuneInSlice(char, g.Questions) {
					g.Questions = append(g.Questions, char)
				}
			}
		}

		return g
	}

	var questionTotal int

	for _, x := range parse(instr) {
		questionTotal += len(newGroup(x).Questions)
	}

	return questionTotal
}
